File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int buy=Integer.MAX_VALUE;
        int max=0;
        for (int i=0;i<prices.length;i++){
            if (prices[i]<buy){
                buy=prices[i];
            }
            else{
                max=Math.max(prices[i]-buy,max);
            }
            
        }
        return max;
    }
}

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        buy=math.inf
        maxp=0
        for i in range(len(prices)):
            if prices[i]<buy:
                buy=prices[i]
            else:
                maxp=max(prices[i]-buy,maxp)
        return maxp
                